File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        temp=float("-inf")
        cnt=0
        cnt1=0
        nums.sort()
        for i in nums:
            if temp == i:
                continue
            elif temp+1 == i:
                cnt1+=1
            else:
                cnt=max(cnt,cnt1+1)
                cnt1=0
            temp=i
        if len(nums) == 0:
            return 0
        cnt=max(cnt,cnt1+1)
        return cnt
            

File: Data Structures & Algorithms/three-integer-sum/submission-2.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        set<vector<int>>st;
        sort(nums.begin(),nums.end());
        for(int i=0;i<nums.size()-2;++i){
            int sum=nums[i];
            int j=i+1;
            int k=nums.size()-1;
            while(j < k){
                if(sum + nums[j]+nums[k] == 0){
                    vector<int>temp;
                    temp={nums[i],nums[j],nums[k]};
                    st.insert(temp);
                    j++;
                    k--;
                }
                else if(sum+nums[j]+nums[k] > 0){
                    k--;
                }
                else{
                    j++;
                }
            }
        }
        vector<vector<int>>v;
        for(auto ku:st){
            v.push_back(ku);
        }
        return v;
    }
};
